contracts/Strategy.sol

```diff
@@ -256,63 +347,45 @@ contract Strategy is BaseStrategy {
         _depositInvestmentTokenInYVault();
     }
 
-    // Allow address to manage Maker's CDP
-    function grantCdpManagingRightsToUser(address user, bool allow)
-        external
-        onlyGovernance
-    {
-        MakerDaiDelegateLib.allowManagingCdp(cdpId, user, allow);
-    }
-
-    // Allow switching between Uniswap and SushiSwap
-    function switchDex(bool isUniswap) external onlyVaultManagers {
-        if (isUniswap) {
-            router = uniswapRouter;
-        } else {
-            router = sushiswapRouter;
-        }
-    }
-
     // Allow external debt repayment
-    // Attempt to take currentRatio to target c-ratio
-    // Passing zero will repay all debt if possible
-    function emergencyDebtRepayment(uint256 currentRatio)
+    // Allow repayment of an arbitrary amount of MIM in case of an emergency
+    // If `amount` is larger than 0, the function will attempt to just repay using MIM
+    // disregarding straty logic.  This could be helpful if for example yvMIM withdrawals are failing and
+    // we want to do a MIM airdrop and direct debt repayment instead
+    // If `amount` is 0, it will attempt to take currentRatio to target c-ratio
+    // Passing zero in `currentRatio` will repay all debt if possible.
+    function emergencyDebtRepayment(uint256 amount, uint256 currentRatio)
         external
         onlyVaultManagers
     {
-        _repayDebt(currentRatio);
+        if (amount > 0) {
+            _repayInvestmentTokenDebt(amount);
+        } else {
+            _repayDebt(currentRatio);
+        }
     }
 
-    // Allow repayment of an arbitrary amount of Dai without having to
-    // grant access to the CDP in case of an emergency
-    // Difference with `emergencyDebtRepayment` function above is that here we
-    // are short-circuiting all strategy logic and repaying Dai at once
-    // This could be helpful if for example yvDAI withdrawals are failing and
-    // we want to do a Dai airdrop and direct debt repayment instead
-    function repayDebtWithDaiBalance(uint256 amount)
-        external
-        onlyVaultManagers
-    {
-        _repayInvestmentTokenDebt(amount);
-    }
 
-    // ******** OVERRIDEN METHODS FROM BASE CONTRACT ************
+    // ******** OVERRIDDEN METHODS FROM BASE CONTRACT ************
```

Comment:
@newmickymousse believe it or not, correct spelling seems to be overriden.

Reply:
Seriously? Woow, I will change it back